src/psyclone/psyir/backend/sir.py

```diff
@@ -0,0 +1,408 @@
+                        "".format(child.operator.name))
+            else:
+                raise VisitorError(
+                    "gen_stencil unsupported stencil index found '{0}'."
+                    "".format(str(child)))
+        else:
+            raise VisitorError(
+                "gen_stencil unsupported (non-stencil) index found '{0}'."
+                "".format(str(child)))
+    return "[{0}]".format(",".join(dims))
+
+
+class SIRWriter(PSyIRVisitor):
+    '''Implements a PSyIR-to-SIR back end for PSyIR kernel code (not
+    currently PSyIR algorithm code which has its own gen method for
+    generating Fortran).
+
+    :param bool skip_nodes: if skip_nodes is False then an exception \
+    is raised if a visitor method for a PSyIR node has not been \
+    implemented, otherwise the visitor continues, printing out a \
+    representation of the unsupported node. This is an optional \
+    argument which defaults to False.
+    :param indent_string: specifies what to use for indentation. This \
+    is an optional argument that defaults to two spaces.
+    :type indent_string: str or NoneType
+    :param int initial_indent_depth: Specifies how much indentation to \
+    start with. This is an optional argument that defaults to 0.
+
+    '''
+    def __init__(self, skip_nodes=False, indent_string="  ",
+                 initial_indent_depth=0):
+        super(SIRWriter, self).__init__(skip_nodes, indent_string,
+                                        initial_indent_depth)
+        self._field_names = set()
+
+    def node_node(self, node):
+        '''Catch any unsupported nodes, output their class names and continue
+        down the node hierarchy (if skip_node is set to True). This is
+        useful for debugging and differs from the base class
+        implementation of skip_nodes which silently continues. If
+        skip_nodes is set to False then raise an exception if an
+        unsupported node is found.
+
+        :param node: an unsupported PSyIR node.
+        :type node: subclass of :py:class:`psyclone.psyGen.Node`
+
+        :returns: the SIR code as a string.
+        :rtype: str
+
+        :raises VisitorError: if skip_nodes is set to False.
+
+        '''
+        if not self._skip_nodes:
+            raise VisitorError(
+                "Class SIRWriter method node_node(), unsupported node "
+                "found '{0}'".format(type(node)))
+        result = "{0}[ {1} start ]\n".format(self._nindent,
+                                             type(node).__name__)
+        self._depth += 1
+        for child in node.children:
+            result += self._visit(child)
+        self._depth -= 1
+        result += "{0}[ {1} end ]\n".format(self._nindent, type(node).__name__)
+        return result
+
+    def nemoloop_node(self, loop_node):
+        '''Supported NEMO loops are triply nested with particular indices (not
+        yet checked) and should contain a nemokern. If this is not the
```

Comment:
s/nemokern/NemoKern/ and below in the description of `loop_node`

Reply:
Yes, thanks.